modules/programs/gitui/gitui.nix

```
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.gitui;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = [ maintainers.mifom ];
+
+  options.programs.gitui = {
+    enable =
+      mkEnableOption "gitui, blazing fast terminal-ui for git written in rust";
+
+    key_config = mkOption {
+      type = types.str;
+      default = "";
+      example = ''
+        exit: ( code: Char('c'), modifiers: ( bits: 2,),),
+        quit: ( code: Char('q'), modifiers: ( bits: 0,),),
+        exit_popup: ( code: Esc, modifiers: ( bits: 0,),),
+      '';
+      description = ''
+        Key config in Ron file format. It written to
+        <filename>~/.config/gitui/key_config.ron</filename>.
+      '';
+    };
```

```suggestion
    keyConfig = mkOption {
      type = types.either types.path types.lines;
      default = "";
      example = literalExpression '''''
        exit: ( code: Char('c'), modifiers: ( bits: 2,),),
        quit: ( code: Char('q'), modifiers: ( bits: 0,),),
        exit_popup: ( code: Esc, modifiers: ( bits: 0,),),
      ''''';
      description = ''
        Key config in Ron file format. This is written to
        <filename>$XDG_CONFIG_HOME/gitui/key_config.ron</filename>.
      '';
    };
```

We prefer using `camelCase` options unless in a freeform module or something that maps directly to an option.

---

Okay, understood